sdk/storage/storage-queue/package.json

```diff
@@ -118,11 +118,16 @@
   },
   "dependencies": {
     "@azure/abort-controller": "^1.0.0",
-    "@azure/core-http": "^3.0.0",
     "@azure/core-paging": "^1.1.1",
     "@azure/core-tracing": "1.0.0-preview.13",
+    "@azure/core-client": "^1.7.1",
+    "@azure/core-rest-pipeline": "^1.10.1",
+    "@azure/core-http-compat": "^2.0.0",
+    "@azure/core-auth": "^1.4.0",
+    "@azure/core-util": "^1.2.0",
     "@azure/logger": "^1.0.0",
-    "tslib": "^2.2.0"
+    "tslib": "^2.2.0",
+    "@azure/storage-blob": "^12.20.0"
```

Comment:
file-datalake is different in that it actually uses blob client underneath for some operations.

Reply:
@jeremymeng is the idea that we put the common files here and then reference them out of the tree? https://github.com/Azure/azure-sdk-for-js/tree/main/sdk/storage/storage-common/src

I wonder if it might also be worth converting storage-common to being a full package like we have done for keyvault-common recently. This would make it much easier to share code between storage libraries in the future. @EmmaZhu thoughts?

Reply:
> is the idea that we put the common files here and then reference them out of the tree?
no the full published common package is what I meant

Reply:
For now I think we can have a copy in storage-queue.

Reply:
Hm yeah seems like storage-file-share is in the same boat (no direct dep on storage-blob) so having a better story here seems important.

Reply:
Having a storage-common package would have some benefit for the duplicated policies, it may also bring a long dependency chain and cause a bit higher dependency complexity in customer environment.  I'll need to think it through more clearly. I don't expect it as a blocking for migration. If so, maybe we can do the change later to make the change more clean and easy to understand.

Reply:
@EmmaZhu I know we have done the shared source code in the past, but the approach has its own problems and we believe common package approach is better for sharing.  That's why we are now moving to use published common package instead (for example, keyvault-common, communication-common, identity is moving to use identity-common).

Yes I agree that separate PRs for migration and moving to common package are better for separate topics and make review easier

Reply:
@jeremymeng ,  I agree that common package approach would bring more benefits. Let's have another PR for storage-common and maybe we should test some minor case like: customer is using storage-blob and storage-queue which references to different dependencies.  Currently storage-blob and storage-queue already have some same dependencies, this should not bring any new potential issue, just we'd need to test the behavior.